### PR TITLE
feat(avoidance): margin can be set independently for each class

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

The avoidance module has some buffer parameters for avoidance path generation.

- lateral buffer
- longitudinal buffer
- envelope polygon buffer

In this PR, it is able to set these parameters independently for each class (e.g. pedestrian, bus, motorcycle).

:arrow_down: It should be merged before this PR.
https://github.com/autowarefoundation/autoware_launch/pull/286

image

## Related links

https://github.com/autowarefoundation/autoware_launch/pull/286

## Tests performed

```yaml
        pedestrian:
          enable: true
          envelope_buffer_margin: 0.8
          safety_buffer_lateral: 1.0
          safety_buffer_longitudinal: 1.0
```

rviz_screenshot_2023_04_08-09_01_27

```yaml
        bus:
          enable: true
          envelope_buffer_margin: 0.3
          safety_buffer_lateral: 0.7
          safety_buffer_longitudinal: 0.0
```

rviz_screenshot_2023_04_08-09_02_03

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by satoshi-ota** [comment]
Running elaluator test now...

File: planning/behavior_path_planner/src/behavior_path_planner_node.cpp [hunk 1]
```diff
@@ -592,15 +593,83 @@ AvoidanceParameters BehaviorPathPlannerNode::getAvoidanceParam()
...
-    std::string ns = "avoidance.target_object.";
-    p.avoid_car = declare_parameter<bool>(ns + "car");
-    p.avoid_truck = declare_parameter<bool>(ns + "truck");
-    p.avoid_bus = declare_parameter<bool>(ns + "bus");
-    p.avoid_trailer = declare_parameter<bool>(ns + "trailer");
-    p.avoid_unknown = declare_parameter<bool>(ns + "unknown");
-    p.avoid_bicycle = declare_parameter<bool>(ns + "bicycle");
-    p.avoid_motorcycle = declare_parameter<bool>(ns + "motorcycle");
-    p.avoid_pedestrian = declare_parameter<bool>(ns + "pedestrian");
+    std::string ns = "avoidance.target_object.motorcycle.";
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/src/behavior_path_planner_node.cpp:596]: @satoshi-ota I love this improvement. Nice. 

Suggest using lambda or function to get rid of duplicated implementations.

**Comment by satoshi-ota** [comment]
[[TIER IV INTERNAL LINK]](https://evaluation.tier4.jp/evaluation/reports/89312be1-7113-54bd-9858-fce4cfbb0757?project_id=prd_jt) There's no critical degration.

File: planning/behavior_path_planner/src/behavior_path_planner_node.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/behavior_path_planner_node.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/3002/commits/0e172b3532fbf210f540915d7af90839b1bd9c16.

**Comment by satoshi-ota** [comment]
@TakaHoribe Could you check https://github.com/autowarefoundation/autoware_launch/pull/286 with this PR?

**Pull request merged**